### PR TITLE
Fix ada.o object build

### LINKED ISSUE #49: Allow `pip install --no-binary` to work
Noted in https://github.com/ada-url/ada-python/pull/46:
> `pip install ada_url --no-binary ada_url` will fail (on both Linux and Windows) because of the absence of `ada.o`

We would need to move the step that builds `ada.o` into the `python -m build` step. I looked into doing previously, but couldn't figure out a portable method.

### PR DESCRIPTION
**Pull request by jameshilliard**
Should fix #49.

### CODE REVIEW

**Comment by jameshilliard** [comment]
I think tests should now all be passing.

**Review by anonrig — APPROVED** [review]

**Comment by bbayles** [comment]
Hmm, this breaks on my machine:

```
ada_url/ada.cpp:11791:31: error: 'value' is unavailable: introduced in macOS 10.13
    helpers::encode_json(hash.value(), back);
                              ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/optional:1057:33: note: 'value' has been explicitly marked unavailable here
    constexpr value_type const& value() const&
                                ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

Any ideas?

**Comment by jameshilliard** [comment]
> introduced in macOS 10.13

Looks like you need to set the [MACOSX_DEPLOYMENT_TARGET](https://github.com/ada-url/ada-python/blob/3c27c3c2faa371c06eda79678d2529c463df4aed/pyproject.toml#L80).

File: pyproject.toml [hunk 1]
```diff
@@ -73,13 +73,15 @@ build = [
...
 environment = { MACOSX_DEPLOYMENT_TARGET="10.15" }
-before-build = "make clean && make c_lib"
+before-build = "make clean"
 
 [tool.cibuildwheel.windows]
 archs = ["AMD64"]
-before-build = '"C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Auxiliary\\Build\\vcvars64.bat" && cl "ada_url\\ada.cpp" /c /nologo /Fo"ada_url\\ada.o" /O2 /GL /MD /W3 /EHsc /std:c++17'
+
+[[tool.cibuildwheel.overrides]]
+select = "pp39-win_amd64"
```
Comment (bbayles) [inline pyproject.toml:86]: Could you leave a comment about why this one is special?

Comment (jameshilliard) [reply pyproject.toml:87]: Added a comment with link to https://github.com/pypy/pypy/issues/5027

**Pull request merged**